symphony-bdk-core/src/main/java/com/symphony/bdk/core/util/ConsumerWithThrowable.java

```diff
@@ -0,0 +1,12 @@
+package com.symphony.bdk.core.util;
+
+import com.symphony.bdk.core.api.invoker.ApiException;
+
+/**
+ * Functional interface which consumes an {@link ApiException} and may throw a {@link Throwable}.
+ * This is used to specify recovery functions in {@link ConsumerWithThrowable}.
+ */
+@FunctionalInterface
+public interface ConsumerWithThrowable {
+  void consume(ApiException e) throws Throwable;
+}
```

Comment:
Could we make it more generic? 

```java
public interface ConsumerWithThrowable<E extends Throwable> {

  void consume(E e) throws Throwable;
}
```

Reply:
Yes we could but if we want to have a fully generic retry mechanism, we will have to add a type parameter for the exception in the RetryWithRecovery class. We cannot catch a type parameter in a try/catch clause. And instanceof works bad with parametrized type : https://stackoverflow.com/questions/1570073/java-instanceof-and-generics